src/devices/src/virtio/mmio.rs

```diff
@@ -304,20 +195,28 @@ impl MmioDevice {
             }
             _ if status == 0 => {
                 if self.device_activated {
-                    match self.device.reset() {
-                        Some((_interrupt_evt, mut queue_evts)) => {
-                            self.device_activated = false;
-                            self.queue_evts.append(&mut queue_evts);
+                    let mut device_activated = true;
+                    let mut device_status = self.device_status;
+
+                    match self.locked_device().reset() {
+                        Some((_interrupt_evt, mut _queue_evts)) => {
+                            device_activated = false;
                         }
-                        // Backend device driver doesn't support reset,
-                        // just mark the device as FAILED.
+
                         None => {
-                            self.device_status |= FAILED;
-                            return;
+                            device_status |= FAILED;
                         }
                     }
+
+                    self.device_activated = device_activated;
+                    self.device_status = device_status;
+                }
+
+                // If the backend device driver doesn't support reset,
+                // just leave the device marked as FAILED.
```

Comment:
nit: With this new code flow this comment is confusing, it took me a bit to validate the logic and the confusion came from this comment which is correct but the wording can be misleading.

I suggest changing it to:
```
// The transport is marked as FAILED in case the backend device doesn't support reset.
// In such a case, skip resetting the transport and just leave it marked as FAILED.
```

Reply:
fixed.